### ISSUE TITLE
Trying to deploy to an existing cluster, recieving an 'unbound variable' error 

### ISSUE THREAD

**Issue by lhatfield** [buildkite-plugins/ecs-deploy-buildkite-plugin#81]
Full error: /var/lib/buildkite-agent/plugins/github-com-buildkite-plugins-ecs-deploy-buildkite-plugin-v2-0-0/hooks/command: line 136: env_vars[@]: unbound variable

```
  - label: ":ecs: :rocket:"

    concurrency_group: "my-service-deploy"
    
    concurrency: 1
    
    plugins:
    
      - ecs-deploy#v1.4.1:
      
          cluster: "<existing cluster>"
          
          service: "<non-existent service>"
          
          task-definition: "deploy/containers.json"
          
          task-family: "<task family>"
          
          image: "<arn>/<image>:latest"
          
          target-group: "<existing targetgroup-arn>"
          
          target-container-port: 3000
          
```
containers.json

```


[
{ 
"essential": true,
    
    "image": "<taskdef arn>",
    
    "memory": 1024,
    
    "name": "sample",
    
    "portMappings": [
    
      {
        "containerPort": 3000,
        
        "hostPort": 3000
        
      }
      
    ]
  }
]

```

I have tried pointing to a different service without a target group and ALB in front of it and still get the same sort of issue. I think Im missing some information on the launching of the fargate containers.

**Comment by ouranos** [comment]
I had the same issue with `v2.0.0`, I got rid of it by setting environment variables in this format:
```yaml
  - label: ":ecs: :rocket:"
    plugins:
      - ecs-deploy#v2.0.0:
          cluster: "<existing cluster>"
          service: "<non-existent service>"
          task-family: "<task family>"
          image: "<arn>/<image>:latest"
          container-definitions: "deploy/containers.json"
          env:
            - "FOO=BAR"
```

**Comment by toote** [comment]
Took me quite a while to find exactly what could have been going on as I couldn't reproduce at all. Once I gave up trying to find the correct incantation of options for the plugin to trigger it (assuming there was a bug in the code and possibly some other option you were using that I was not seeing) I found the explanation [about arrays being reported as undefined variables](https://stackoverflow.com/a/28058737/1352026).

Basically, this is only an issue when the environment variables to set in containers is empty and only when you are using Bash version 4.3 (thanks @ouranos for the reply that helped me narrow down possible causes).

I will close this issue for the time being but will make a note about it in the documentation in a future release.
